src/adapters/Cornerstone3D/EllipticalROI.js

```diff
@@ -24,60 +30,74 @@ class EllipticalROI {
         // ellipse so we need to identify if the majorAxis is horizontal or vertical
-        // and then choose the correct points to use for the ellipse.
-
-        const majorAxisStart = { x: GraphicData[0], y: GraphicData[1] };
-        const majorAxisEnd = { x: GraphicData[2], y: GraphicData[3] };
-        const minorAxisStart = { x: GraphicData[4], y: GraphicData[5] };
-        const minorAxisEnd = { x: GraphicData[6], y: GraphicData[7] };
-
-        const majorAxisIsHorizontal = majorAxisStart.y === majorAxisEnd.y;
-        const majorAxisIsVertical = majorAxisStart.x === majorAxisEnd.x;
-
-        let ellipsePointsImage;
-
-        if (majorAxisIsVertical) {
-            ellipsePointsImage = [
-                majorAxisStart.x,
-                majorAxisStart.y,
-                majorAxisEnd.x,
-                majorAxisEnd.y,
-                minorAxisStart.x,
-                minorAxisStart.y,
-                minorAxisEnd.x,
-                minorAxisEnd.y
-            ];
-        } else if (majorAxisIsHorizontal) {
-            ellipsePointsImage = [
-                minorAxisStart.x,
-                minorAxisStart.y,
-                minorAxisEnd.x,
-                minorAxisEnd.y,
-                majorAxisStart.x,
-                majorAxisStart.y,
-                majorAxisEnd.x,
-                majorAxisEnd.y
-            ];
-        } else {
-            throw new Error("ROTATED ELLIPSE NOT YET SUPPORTED");
-        }
-
-        const points = [];
-        for (let i = 0; i < ellipsePointsImage.length; i += 2) {
+        // in the image plane and then choose the correct points to use for the ellipse.
+        const pointsWorld = [];
+        for (let i = 0; i < GraphicData.length; i += 2) {
             const worldPos = imageToWorldCoords(imageId, [
-                ellipsePointsImage[i],
-                ellipsePointsImage[i + 1]
+                GraphicData[i],
+                GraphicData[i + 1]
             ]);
 
-            points.push(worldPos);
+            pointsWorld.push(worldPos);
+        }
+
+        const majorAxisStart = vec3.fromValues(...pointsWorld[0]);
+        const majorAxisEnd = vec3.fromValues(...pointsWorld[1]);
+        const minorAxisStart = vec3.fromValues(...pointsWorld[2]);
+        const minorAxisEnd = vec3.fromValues(...pointsWorld[3]);
+
+        const majorAxisVec = vec3.create();
+        vec3.sub(majorAxisVec, majorAxisEnd, majorAxisStart);
+
+        const minorAxisVec = vec3.create();
+        vec3.sub(minorAxisVec, minorAxisEnd, minorAxisStart);
+
+        const imagePlaneModule = metadata.get("imagePlaneModule", imageId);
+
+        if (!imagePlaneModule) {
+            throw new Error("imageId does not have imagePlaneModule metadata");
+        }
+
+        const { columnCosines } = imagePlaneModule;
+
+        // find which axis is parallel to the columnCosines
+        const columnCosinesVec = vec3.fromValues(...columnCosines);
+
+        const projectedMajorAxisOnColVec = vec3.dot(
```

Comment:
I think it would be safer to normalize these vectors: `majorAxisVec` and `minorAxisVec`.

Later we just compare them to EPSILON, which is not a fair comparison if they aren't unit vectors. If the ellipse is really big, the comparison to ~0 could break.